contracts/NodeDelegator.sol

```diff
@@ -267,22 +314,41 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
         _unpause();
     }
 
-    /// @dev allow NodeDelegator to receive ETH
-    function sendETHFromDepositPoolToNDC() external payable override {
-        // only allow LRT deposit pool to send ETH to this contract
-        address lrtDepositPool = lrtConfig.getContract(LRTConstants.LRT_DEPOSIT_POOL);
-        if (msg.sender != lrtDepositPool) {
-            revert InvalidETHSender();
-        }
-
-        emit ETHDepositFromDepositPool(msg.value);
-    }
-
     /// @dev opts in for rebase so the asset's token balance will increase
     function optIn(address asset) external onlyLRTAdmin onlySupportedAsset(asset) {
         IOETH(asset).rebaseOptIn();
     }
 
+    /// @dev Approves the SSV Network contract to transfer SSV tokens for deposits
+    function approveSSV() external onlyLRTManager {
+        address SSV_TOKEN_ADDRESS = lrtConfig.getContract(LRTConstants.SSV_TOKEN);
+        address SSV_NETWORK_ADDRESS = lrtConfig.getContract(LRTConstants.SSV_NETWORK);
+
+        IERC20(SSV_TOKEN_ADDRESS).approve(SSV_NETWORK_ADDRESS, type(uint256).max);
+    }
+
+    /// @dev Deposits more SSV Tokens to the SSV Network contract which is used to pay the SSV Operators
+    function depositSSV(uint64[] memory operatorIds, uint256 amount, Cluster memory cluster) external onlyLRTManager {
+        address SSV_NETWORK_ADDRESS = lrtConfig.getContract(LRTConstants.SSV_NETWORK);
+
+        ISSVNetwork(SSV_NETWORK_ADDRESS).deposit(address(this), operatorIds, amount, cluster);
+    }
+
+    function registerSsvValidator(
+        bytes calldata publicKey,
+        uint64[] calldata operatorIds,
+        bytes calldata sharesData,
+        uint256 amount,
+        Cluster calldata cluster
+    )
+        external
+        onlyLRTOperator
+    {
+        address SSV_NETWORK_ADDRESS = lrtConfig.getContract(LRTConstants.SSV_NETWORK);
+
+        ISSVNetwork(SSV_NETWORK_ADDRESS).registerValidator(publicKey, operatorIds, sharesData, amount, cluster);
+    }
+
     /// @dev allow NodeDelegator to receive ETH rewards
     receive() external payable {
         emit ETHRewardsReceived(msg.value);
```

Comment:
I think this event is going to be fired every time we unwrap WETH?

Reply:
That's correct. I suspect it'll also get confused with validator withdraws. I'll remove the `ETHRewardsReceived` event rather than adding checking the msg.sender is not the WETH contract.

Reply:
Agreed.